colour/utilities/tests/test_array.py

```diff
@@ -1116,6 +1124,95 @@ def test_from_range_int(self):
             )
 
 
+class TestIsNdarrayCopyEnabled(unittest.TestCase):
+    """
+    Define :func:`colour.utilities.array.is_ndarray_copy_enabled` definition
+    unit tests methods.
+    """
+
+    def test_is_ndarray_copy_enabled(self):
+        """
+        Test :func:`colour.utilities.array.is_ndarray_copy_enabled` definition.
+        """
+
+        with ndarray_copy_enable(True):
+            self.assertTrue(is_ndarray_copy_enabled())
+
+        with ndarray_copy_enable(False):
+            self.assertFalse(is_ndarray_copy_enabled())
+
+
+class TestSetNdarrayCopyEnabled(unittest.TestCase):
+    """
+    Define :func:`colour.utilities.array.set_ndarray_copy_enable` definition
+    unit tests methods.
+    """
+
+    def test_set_ndarray_copy_enable(self):
+        """
+        Test :func:`colour.utilities.array.set_ndarray_copy_enable` definition.
+        """
+
+        with ndarray_copy_enable(is_ndarray_copy_enabled()):
+            set_ndarray_copy_enable(True)
+            self.assertTrue(is_ndarray_copy_enabled())
+
+        with ndarray_copy_enable(is_ndarray_copy_enabled()):
+            set_ndarray_copy_enable(False)
+            self.assertFalse(is_ndarray_copy_enabled())
+
+
+class TestNdarrayCopyEnable(unittest.TestCase):
+    """
+    Define :func:`colour.utilities.array.ndarray_copy_enable` definition unit
+    tests methods.
+    """
+
+    def test_ndarray_copy_enable(self):
+        """
+        Test :func:`colour.utilities.array.ndarray_copy_enable` definition.
+        """
+
+        with ndarray_copy_enable(True):
+            self.assertTrue(is_ndarray_copy_enabled())
+
+        with ndarray_copy_enable(False):
+            self.assertFalse(is_ndarray_copy_enabled())
+
+        @ndarray_copy_enable(True)
+        def fn_a():
+            """:func:`ndarray_copy_enable` unit tests :func:`fn_a` definition."""
+
+            self.assertTrue(is_ndarray_copy_enabled())
+
+        fn_a()
+
+        @ndarray_copy_enable(False)
+        def fn_b():
+            """:func:`ndarray_copy_enable` unit tests :func:`fn_b` definition."""
+
+            self.assertFalse(is_ndarray_copy_enabled())
+
+        fn_b()
+
+
+class TestNdarrayCopy(unittest.TestCase):
+    """
+    Define :func:`colour.utilities.array.ndarray_copy` definition unit
+    tests methods.
+    """
+
+    def test_ndarray_copy(self):
+        """Test :func:`colour.utilities.array.ndarray_copy` definition."""
+
+        a = np.linspace(0, 1, 10)
+        with ndarray_copy_enable(True):
+            self.assertNotEqual(id(ndarray_copy(a)), id(a))
+
+        with ndarray_copy_enable(False):
+            self.assertEqual(id(ndarray_copy(a)), id(a))
+
+
```

Comment:
We should test the other decorating modes as well. This leaves the function decorator untested.

Reply:
The decorator is tested here: https://github.com/colour-science/colour/pull/1151/files/27f6f4b79513ab2f00a2dbd09e15ecead8b47e91#diff-a99802bce9e534d4136fb2c2f7d425fdc96e4f19a6505769437c39512f2a983dR1182